pkg/cron/manager.go

```diff
@@ -0,0 +1,59 @@
+package cron
+
+import (
+	"fmt"
+	"log/slog"
+
+	"github.com/robfig/cron/v3"
+)
+
+var logger *slog.Logger
+
+func getLogger() *slog.Logger {
+	if logger == nil {
+		logger = slog.Default().With(slog.Any("src", "cronManager"))
+	}
+	return logger
+}
+
+type Doer func() error
+
+type Manager struct {
+	cron *cron.Cron
+}
+
+func NewManager() *Manager {
+	return &Manager{
+		cron: cron.New(),
+	}
+}
+
+func (m *Manager) intercept(f Doer) func() {
+	return func() {
+		if err := f(); err != nil {
+			getLogger().Error("cron job failed", slog.Any("error", err), slog.Any("job", f))
+			return
+		}
+		getLogger().Debug("cron job done successfully", slog.Any("job", f))
+	}
+}
+
+func (m *Manager) AddJob(spec string, f Doer) error {
+	_, err := m.cron.AddFunc(spec, m.intercept(f))
+	if err != nil {
+		getLogger().Error("failed to add cron job", slog.Any("error", err))
```

Comment:
Gotcha. That's exactly what I mean in other comments. You are login error returned from the method on the [higher level](https://github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pull/11/files#diff-7b1256d8c85d680a082bdf71358641cb599003f427d8604bb13511af67c221d1R87). So why are you logging it also here? With the same message